ddbToEsDlqLambda/index.ts

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable no-await-in-loop */
+import AWS from 'aws-sdk';
+import getComponentLogger from './loggerBuilder';
+
+const logger = getComponentLogger();
+const { QUEUE_URL, DDB_TO_ES_LAMBDA_NAME } = process.env;
+
+const sqs = new AWS.SQS();
+const lambda = new AWS.Lambda();
+const dynamodbstreams = new AWS.DynamoDBStreams();
+
+const getMessages = async (queueUrl) => {
+    const resp = await sqs
+        .receiveMessage({
+            QueueUrl: queueUrl,
+            MaxNumberOfMessages: 10,
+            VisibilityTimeout: 300,
+            WaitTimeSeconds: 0,
+        })
+        .promise();
+
+    return resp.Messages;
+};
+
+const deleteMessages = async (queueUrl, messagesToDelete) => {
+    await sqs
+        .deleteMessageBatch({
+            QueueUrl: queueUrl,
+            Entries: messagesToDelete,
+        })
+        .promise();
+};
+
+const invokeDdbToEsLambda = async (records) => {
+    await lambda
+        .invoke({
+            FunctionName: DDB_TO_ES_LAMBDA_NAME,
+            InvocationType: 'RequestResponse',
+            Payload: JSON.stringify(records),
+        })
+        .promise();
+};
+
+const getRecordsFromDbStream = async (message) => {
+    const resp = await dynamodbstreams
+        .getShardIterator({
+            ShardId: message.DDBStreamBatchInfo.shardId,
+            ShardIteratorType: 'AT_SEQUENCE_NUMBER',
+            StreamArn: message.DDBStreamBatchInfo.streamArn,
+            SequenceNumber: message.DDBStreamBatchInfo.startSequenceNumber,
+        })
+        .promise();
+
+    return dynamodbstreams
+        .getRecords({
+            ShardIterator: resp.ShardIterator,
+        })
+        .promise();
+};
+
+exports.handler = async () => {
+    let messages;
+    try {
+        messages = await getMessages(QUEUE_URL);
+        logger.debug('Recevied messages', JSON.stringify(messages));
+    } catch (e) {
+        logger.error('Failed to receive messages', JSON.stringify(e));
```

Comment:
I think you should consider pushing the try-catch into getMessages, since it is repeated below.